presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PickTableLayout.java

```diff
@@ -250,9 +344,49 @@ public static PlanNode pushPredicateIntoTableScan(
             SqlParser parser,
             ExpressionDomainTranslator domainTranslator)
     {
+        if (metadata.isPushdownFilterSupported(session, node.getTable())) {
+            Map<NodeRef<Expression>, Type> predicateTypes = getExpressionTypes(
+                    session,
+                    metadata,
+                    parser,
+                    types,
+                    predicate,
+                    emptyList(),
+                    WarningCollector.NOOP,
+                    false);
+
+            BiMap<String, String> symbolToColumnNameMap = node.getAssignments().entrySet().stream()
+                    .collect(toImmutableBiMap(
+                            entry -> entry.getKey().getName(),
+                            entry -> getColumnName(session, metadata, node.getTable(), entry.getValue())));
+            RowExpression translatedPredicate = SqlToRowExpressionTranslator.translate(predicate, predicateTypes, ImmutableMap.of(), metadata.getFunctionManager(), metadata.getTypeManager(), session, false)
```

Comment:
Let's see if this can be altered based on @rongrong's comment (https://github.com/prestodb/presto/pull/12875/commits/79b83209b10feaffd8c0cb63b7a51fdda47df160#r289458788).